.ci/magician/cmd/test_terraform_vcr.go

```diff
@@ -292,7 +292,7 @@ func modifiedPackages(newBranch, oldBranch string, rnr ExecRunner) (map[string]s
 	}
 	var goFiles []string
```

Comment:
Could you split the logic from here to the end of the function into a separate function and add thorough unit tests for it?